pydantic_ai_slim/pydantic_ai/models/anthropic.py

```diff
@@ -326,12 +345,21 @@ def _process_response(self, response: BetaMessage) -> ModelResponse:
                     )
                 )
 
+        # Map finish reason from stop_reason
+        mapped_finish = _map_anthropic_finish(getattr(response, 'stop_reason', None))
+        provider_details: dict[str, Any] | None = None
+        raw_finish: str | None = getattr(response, 'stop_reason', None)
```

Comment:
We can move this (without `getattr`) to above `mapped_finish` so we don't read the field twice